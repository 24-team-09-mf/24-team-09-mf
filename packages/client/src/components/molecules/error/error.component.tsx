import { SyntheticEvent } from 'react'
import { useNavigate } from 'react-router-dom'
import { Wrapper, Code, Description, BackLink } from './error.styles'

import { IErrorBlockProps } from './error.types'

export const ErrorBlock: React.FC<IErrorBlockProps> = ({ code, text }) => {
  const navigate = useNavigate()

  const onBack = (e: SyntheticEvent) => {
    e.preventDefault()
    navigate(-1)
  }

  return (
    <Wrapper>
      <Code>{code}</Code>
      <Description>{text}</Description>
      <BackLink href="#" onClick={onBack}>
        Назад
      </BackLink>
    </Wrapper>
  )
}
