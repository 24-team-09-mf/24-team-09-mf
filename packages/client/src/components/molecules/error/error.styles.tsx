import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: auto;
`
export const Code = styled.h3`
  color: #579945;
  font-size: 128px;
  line-height: 100%;
`
export const Description = styled.p`
  color: #579945;
  font-size: 24px;
`
export const BackLink = styled.a`
  color: #333333;
  cursor: pointer;
  font-weight: 500;
  font-size: 16px;
  margin-top: 40px;
`
