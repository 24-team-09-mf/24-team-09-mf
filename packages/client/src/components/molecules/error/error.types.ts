export interface IErrorBlockProps {
  code: string
  text: string
}
