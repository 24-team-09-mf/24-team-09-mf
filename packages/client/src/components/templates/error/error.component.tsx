import { Container } from '@/components/layouts/header/header.styles'
import { ErrorBlock } from '@/components/molecules/error'

export const ErrorTemplate: React.FC = () => {
  return (
    <Container>
      <ErrorBlock code="500" text="Уже чиним" />
    </Container>
  )
}
