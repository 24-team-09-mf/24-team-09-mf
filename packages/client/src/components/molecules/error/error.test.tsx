import * as renderer from 'react-test-renderer'
import { ErrorBlock } from './error.component'

describe('Тест компонента ErrorBlock', () => {
  it('Тестирование снепшотом', () => {
    const component = renderer
      .create(<ErrorBlock code="404" text="sorry" />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
