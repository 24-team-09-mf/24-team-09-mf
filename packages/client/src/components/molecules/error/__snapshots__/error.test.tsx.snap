// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента ErrorBlock Тестирование снепшотом 1`] = `
<div
  className="sc-beqWaB dpjSCD"
>
  <h3
    className="sc-gueYoa dAPtVp"
  >
    404
  </h3>
  <p
    className="sc-dmqHEX bSYZBS"
  >
    sorry
  </p>
  <a
    className="sc-hLseeU fznFxu"
    onClick={[Function]}
  >
    Назад
  </a>
</div>
`;
